src/scroll-timeline-css.js

```diff
@@ -144,36 +144,20 @@ export function initCSSPolyfill() {
 
   initMutationObserver();
 
-  // Cache all Proxy Animations
-  let proxyAnimations = new WeakMap();
-
   // We are not wrapping capturing 'animationstart' by a 'load' event,
   // because we may lose some of the 'animationstart' events by the time 'load' is completed.
   window.addEventListener('animationstart', (evt) => {
     evt.target.getAnimations().filter(anim => anim.animationName === evt.animationName).forEach(anim => {
-      // Create a per-element cache
-      if (!proxyAnimations.has(evt.target)) {
-        proxyAnimations.set(evt.target, new Map());
-      }
-      const elementProxyAnimations = proxyAnimations.get(evt.target);
-
-      // Store Proxy Animation in the cache
-      if (!elementProxyAnimations.has(anim.animationName)) {
-        const result = createScrollTimeline(anim, anim.animationName, evt.target);
-        if (result && result.timeline && anim.timeline != result.timeline) {
-          elementProxyAnimations.set(anim.animationName, new ProxyAnimation(anim, result.timeline, result.animOptions));
-        } else {
-          elementProxyAnimations.set(anim.animationName, null);
-        }
-      }
-      
-      // Get Proxy Animation from cache
-      const proxyAnimation = elementProxyAnimations.get(anim.animationName);
-
-      // Swap the original animation with the proxied one
-      if (proxyAnimation !== null) {
+      const result = createScrollTimeline(anim, anim.animationName, evt.target);
+      // If the CSS Animation refers to a scroll or view timeline we need to proxy the animation instance.
+      if (result.timeline && !(anim instanceof ProxyAnimation)) {
+        const proxyAnimation = new ProxyAnimation(anim, result.timeline, result.animOptions);
         anim.pause();
         proxyAnimation.play();
+      } else {
+        // If the timeline was removed or the animation was already an instance of a proxy animation,
+        // invoke the set the timeline procedure on the existing animation.
+        anim.timeline = result.timeline;
```

Comment:
Don’t see why this needs to be set every time. If it’s not needed, then we can return earlier in this block, after doing a simple `anim instanceof ProxyAnimation` check, thereby preventing excessive calls to `createScrollTimeline`.

Reply:
If the style changed such that the underlying CSS animation was restarted there's a good chance that the timeline may have changed as well. We could try removing this but I don't expect this to occur frequently. Scroll driven animations tend to start once and run for the lifetime of their element / the page.